### PR TITLE
Add options to disable SO_REUSEPORT of grpc.

### PR DESCRIPTION
**Pull request by gongweibao**
Some old kernel doesn't support SO_REUSEPORT feature.

https://github.com/tensorflow/tensorflow/issues/9377
https://github.com/PaddlePaddle/Paddle/issues/12111

### CODE REVIEW

**Comment by jacquesqiao** [comment]
what is the purpose of reuse_port for grpc?

File: paddle/fluid/framework/parallel_executor.cc [hunk 1]
```diff
@@ -218,8 +218,9 @@ void ParallelExecutor::BCastParamsToDevices(
         platform::NCCLGroupGuard guard;
         for (size_t i = 0; i < member_->places_.size(); ++i) {
           auto &nccl_ctx = member_->nccl_ctxs_->at(member_->places_[i]);
-          platform::dynload::ncclBcast(buffers[i], numel, data_type, 0,
-                                       nccl_ctx.comm_, nccl_ctx.stream());
+          PADDLE_ENFORCE(
```
Comment (typhoonzero) [inline paddle/fluid/framework/parallel_executor.cc]: CUDNN_ENFORCE(...)

Comment (gongweibao) [reply paddle/fluid/framework/parallel_executor.cc]: Sorry, it should not be fixed in this PR! So I clean it.

**Comment by gongweibao** [comment]
@jacquesqiao Old kernel such like 2.6.32 doesn't support SO_REQUEST feature. And TensorFlow disables it default :https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc#L191

**Comment by gongweibao** [comment]
http://www.blogjava.net/yongboy/archive/2015/02/12/422893.html @jacquesqiao

**Pull request merged**